tools/integration_tests/read_large_files/concurrent_read_files_test.go

```diff
@@ -68,16 +68,23 @@ func TestReadFilesConcurrently(t *testing.T) {
 		createFileOnDiskAndCopyToMntDir(fileInLocalDisk, file, FiveHundredMB, t)
 	}
 
-	// For waiting on threads.
-	var wg sync.WaitGroup
+	var eG errgroup.Group
 
 	for i := 0; i < NumberOfFilesInLocalDiskForConcurrentRead; i++ {
-		// Increment the WaitGroup counter.
-		wg.Add(1)
-		// Thread to read file.
-		go readFile(filesPathInLocalDisk[i], filesPathInMntDir[i], &wg, t)
+		// Copy the current value of i into a local variable to avoid data races.
+		fileIndex := i
+
+		// Thread to read the current file.
+		eG.Go(func() error {
+			// Reduce thread count when it is done.
```

Comment:
Same here, what does this comment mean?
Can we directly return, instead of storing the error first?